src/main/java/org/deepsymmetry/beatlink/dbserver/ConnectionManager.java

```diff
@@ -205,8 +206,10 @@ public <T> T invokeWithClientSession(int targetPlayer, ClientTask<T> task, Strin
 
     /**
      * Keeps track of the database server ports of all the players we have seen on the network.
+     * They are grouped by IP address, because players in compound devices like the XDJ-XZ and
+     * XDJ-AZ share a single dbserver instance.
      */
-    private final Map<Integer, Integer> dbServerPorts = new ConcurrentHashMap<>();
+    private final Map<InetAddress, Integer> dbServerPorts = new ConcurrentHashMap<>();
```

Comment:
This is the main change I mentioned on Zulip. We now track database servers by IP address, rather than by device number, because that is how they are actually partitioned.